.github/workflows/cmake.yml

```diff
@@ -409,13 +378,37 @@ jobs:
         choco install visualstudio2017community
         choco install visualstudio2017-workload-nativedesktop
         choco install visualstudio2017-workload-vctools
+        
+    - name: Install MinGW
+      id: install_msys2
+      if: (startsWith(matrix.config.os, 'windows-2019') && contains(matrix.config.mingw, 'MINGW'))
+      shell: msys2 {0}
+      working-directory: ${{ env.HOME }}
+      run: |
+          pacman -S --noconfirm mingw-w64-${{ matrix.config.mingw-arch }}-cmake \
+          mingw-w64-${{ matrix.config.mingw-arch }}-ninja \
+          mingw-w64-${{ matrix.config.mingw-arch }}-${{ matrix.config.cxx }} \
+          mingw-w64-${{ matrix.config.mingw-arch }}-toolchain
+
+          cmake $GITHUB_WORKSPACE \
+          -DCMAKE_BUILD_TYPE=${{ matrix.config.buildtype }} \
+          -DCMAKE_CXX_STANDARD=${{ matrix.config.cxxver }} \
+          -DBUILD_TESTING=ON \
+          -DCMAKE_CXX_FLAGS=${{ matrix.config.cxx_flags }} \
+          -DCMAKE_EXE_LINKER_FLAGS=${{ matrix.config.exe_linker_flags }} \
+          -DCMAKE_VERBOSE_MAKEFILE=ON
+
+          cmake --build . --config ${{ matrix.config.buildtype }}
+          
+          ctest --output-on-failure -C ${{ matrix.config.buildtype }}
 
     - name: Create Build Environment
       # Some projects don't allow in-source building, so create a separate build directory
       # We'll use this as our working directory for all subsequent commands
       run: cmake -E make_directory ${{runner.workspace}}/build
 
     - name: Configure CMake
+      if: (!contains(matrix.config.mingw, 'MINGW'))
       # Use a bash shell so we can use the same syntax for environment variable
       # access regardless of the host operating system
       shell: bash
```

Comment:
could the configure for mingw be done here if we used something like

shell: ${{ matrix.config.mingw == 'MINGW64' && 'msys2 {0}' || 'bash' }}

like in this example: https://docs.github.com/en/actions/learn-github-actions/expressions#example

Reply:
ok

Reply:
@andreasbuhr ,
```bash
The workflow is not valid. .github/workflows/cmake.yml (Line: 466, Col: 14): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.config.mingw == 'MINGW64' && 'msys2 {0}' || 'bash'
```

Reply:
my branch test: https://github.com/kassane/cppcoro/commit/010e42cc95c26331f2ebd76f708cb404d4f81d15

Reply:
I did some googling and it seems there is no easy way to configure the shell using the matrix. I just found solutions with wrapper shells.

In that case I'd say we go with your solution to do everything in one step. Could you rename the step to something which represents its contents, like "configure, compile and run in msys" or similar.